### PR TITLE
Improvement: Multiple changes to DefaultLayout/ElkLayout/multiple elk changes

### PR DESCRIPTION
**Pull request by thibaudlabat**
- Move ELK parameters from ElkLayout to DefautConfiguration.
- Add `containerId` optional parameter to `arrangeComponentsPosition` prototype in `DefaultLayout` and `ElkLayout` to allow repositioning of a container's children.

### CODE REVIEW

File: src/draw/ElkLayout.js [hunk 1]
```diff
@@ -301,6 +305,8 @@ class ElkLayout extends DefaultLayout {
     const currentDepth = parentNode.depth + 1;
 
     return allLinks
+    // We ignore links for nodes that we don't know about.
```
Comment (Zorin95670) [inline src/draw/ElkLayout.js:308]: `Ignore links that's not related to given container`

File: src/draw/DefaultLayout.js [hunk 2]
```diff
@@ -20,9 +20,12 @@ class DefaultLayout {
   /**
    * Does nothing to components.
    * One can override this method to implement an algorithm for automatic components layout.
+   * @param {string} [containerId] - Container of which we must arrange the children. If not given,
```
Comment (Zorin95670) [inline src/draw/DefaultLayout.js:23]: `The container within which we need to organize the children, and if not specified, all components will be reorganized.` (for all)

File: cypress/e2e/automatic-layout.feature [hunk 3]
```diff
@@ -5,13 +5,54 @@ Feature: Test automatic layout for the graph
     And I visit the "/"
 
   Scenario: Auomatic layout should have a (deterministic) given result
+    # Initial positions
+    Then I expect "#__shadowRoot" to be at position 0,0
```
Comment (Zorin95670) [inline cypress/e2e/automatic-layout.feature:9]: Split into two scenario, one that check all is rerranged and another to check only component in container is rearranged.

File: tests/unit/draw/DefaultDrawer.spec.js [hunk 4]
```diff
@@ -1085,20 +1085,22 @@ describe('Test Class: DefaultDrawer()', () => {
   });
 
   describe('Test method: arrangeComponentsPosition', () => {
```
Comment (Zorin95670) [inline tests/unit/draw/DefaultDrawer.spec.js:1087]: Move all modification of `Update tests` commit with commit that modified DefaultDrawer and other.

File: changelog.md [hunk 5]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+## [Unreleased]
+
+### Added
+
+- Add optional containerId argument to DefaultLayout.rearrangeComponentsPosition
```
Comment (Zorin95670) [inline changelog.md:12]: put DefaultLayout.rearrangeComponentsPosition in `` .
Missing `.` at the end.

File: changelog.md [hunk 6]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
...
+
+### Changed
+
+- Move ELK parameters from ElkLayout to DefautConfiguration
```
Comment (Zorin95670) [inline changelog.md:16]: Missing `.` at the end.

File: changelog.md [hunk 7]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+## [Unreleased]
```
Comment (Zorin95670) [inline changelog.md:8]: Missing `guides/migrations/0.19.0_to_0.X.0.md` modification.

File: changelog.md [hunk 8]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
...
+
```
Comment (Zorin95670) [inline changelog.md:9]: Can you rename in another commit 
`guides/migrations/0.18.0_to_0.X.0.md` to `guides/migrations/0.18.0_to_0.19.0.md`

File: src/draw/ElkLayout.js (hunk 1 above)
Comment (thibaudlabat) [reply src/draw/ElkLayout.js:308]: done

File: src/draw/DefaultLayout.js (hunk 2 above)
Comment (thibaudlabat) [reply src/draw/DefaultLayout.js:23]: done

Comment (thibaudlabat) [reply src/draw/DefaultLayout.js]: done

File: cypress/e2e/automatic-layout.feature (hunk 3 above)
Comment (thibaudlabat) [reply cypress/e2e/automatic-layout.feature:9]: done

File: src/draw/ElkLayout.js (hunk 1 above)
Comment (thibaudlabat) [reply src/draw/ElkLayout.js]: done

File: tests/unit/draw/DefaultDrawer.spec.js (hunk 4 above)
Comment (thibaudlabat) [reply tests/unit/draw/DefaultDrawer.spec.js:1087]: done

File: changelog.md (hunk 5 above)
Comment (thibaudlabat) [reply changelog.md:12]: done

File: changelog.md (hunk 6 above)
Comment (thibaudlabat) [reply changelog.md:16]: done

File: changelog.md (hunk 8 above)
Comment (thibaudlabat) [reply changelog.md:9]: done

File: changelog.md (hunk 7 above)
Comment (thibaudlabat) [reply changelog.md:8]: done

File: demo/src/App.vue [hunk 9]
```diff
@@ -25,19 +25,20 @@
...
     </div>
+    <div id="fieldset-container">
     <fieldset>
       <legend>Rename a component ID</legend>
       <label>Choose an ID:</label>
       <select
-        v-model="selectedId"
-        name="ids"
-        id="component-id"
+          v-model="selectedId"
```
Comment (Zorin95670) [inline demo/src/App.vue:33]: bad indentation, fix it for all tab in template

File: demo/src/App.vue [hunk 10]
```diff
@@ -48,6 +49,28 @@
...
+      <label>Parent component:</label>
+        <select
+            v-model="selectedAutoLayoutId"
+            name="idsAutomaticLayout"
+            id="automatic-layout-children-input"
+        >
+          <option value="">(Reorganize everything)</option>
+          <option
+              v-for="id in componentsIds"
+              :key="id"
```
Comment (Zorin95670) [inline demo/src/App.vue:64]: key are already use, please make it unique

File: demo/src/App.vue [hunk 11]
```diff
@@ -48,6 +49,28 @@
...
+              :value="id"
+          >
+            {{ id }}
+          </option>
+        </select>
+      <br/>
+      <button @click="automaticLayoutOfChildren" id="automatic-layout-children-button">Automatic layout</button>
```
Comment (Zorin95670) [inline demo/src/App.vue:71]: Automatic layout /> Rearrange

File: demo/src/App.vue [hunk 12]
```diff
@@ -48,6 +49,28 @@
...
+
+    <fieldset>
+      <legend>Rearrange components</legend>
+      <label>Parent component:</label>
+        <select
+            v-model="selectedAutoLayoutId"
+            name="idsAutomaticLayout"
+            id="automatic-layout-children-input"
+        >
+          <option value="">(Reorganize everything)</option>
```
Comment (Zorin95670) [inline demo/src/App.vue:61]: Reorganize -> Rearrange

File: src/models/DefaultConfiguration.js [hunk 13]
```diff
@@ -51,6 +54,25 @@ class DefaultConfiguration {
...
      */
     this.tags = props.tags || [];
+    /**
+     Parameters for the ELK automatic layout system.
+     @see https://eclipse.dev/elk/reference/options.html
+     @type {object}
+     */
+    this.elkParams = {
+      // default parameters
+      'elk.algorithm': 'elk.layered',
```
Comment (Zorin95670) [inline src/models/DefaultConfiguration.js:64]: put all default into configuration

File: src/models/DefaultConfiguration.js [hunk 14]
```diff
@@ -51,6 +54,25 @@ class DefaultConfiguration {
      * @default []
      */
     this.tags = props.tags || [];
+    /**
+     Parameters for the ELK automatic layout system.
+     @see https://eclipse.dev/elk/reference/options.html
+     @type {object}
+     */
+    this.elkParams = {
```
Comment (Zorin95670) [inline src/models/DefaultConfiguration.js:62]: Missing test constructor in DefaultConfiguration.spec.js

File: demo/src/App.vue (hunk 10 above)
Comment (thibaudlabat) [reply demo/src/App.vue:64]: done

File: demo/src/App.vue (hunk 11 above)
Comment (thibaudlabat) [reply demo/src/App.vue:71]: done

File: demo/src/App.vue (hunk 12 above)
Comment (thibaudlabat) [reply demo/src/App.vue:61]: done

File: demo/src/App.vue (hunk 9 above)
Comment (thibaudlabat) [reply demo/src/App.vue:33]: done

File: src/models/DefaultConfiguration.js (hunk 14 above)
Comment (thibaudlabat) [reply src/models/DefaultConfiguration.js:62]: done

File: tests/unit/models/DefaultConfiguration.spec.js [hunk 15]
```diff
@@ -39,5 +40,17 @@ describe('Test class: DefaultConfiguration', () => {
       expect(config.defaultFileExtension).toEqual('.test');
       expect(config.tags).toEqual(['test']);
     });
+
+    it('initializes ELK configuration properly', () => {
```
Comment (Zorin95670) [inline tests/unit/models/DefaultConfiguration.spec.js:44]: add expect in other test instead of adding it.
Like:
`expect(config.elkParams).toEqual({...})`

Comment (thibaudlabat) [reply tests/unit/models/DefaultConfiguration.spec.js:44]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**